### ISSUE TITLE
Failing on Big Sur with Apple Silicon

### ISSUE THREAD

**Issue by gordonbazeley** [nriley/brightness#34]
Hi - I've just got a new MacBook Pro running on Big Sur with Apple Silicon.  The brightness command is now failing e.g.

brightness -l
display 0: main, active, awake, online, external, ID 0x2
brightness: failed to get brightness of display 0x2 (error -536870201)
display 1: active, awake, online, built-in, ID 0x1
brightness: failed to get brightness of display 0x1 (error -536870201)

I'm not sure whether this is down to Big Sur or Apple Silicon - any assistance would be appreciated

**Comment by nriley** [comment]
Sorry, no Apple Silicon Mac here probably until next year.  Do you have 2 displays or is it finding the Touch Bar?

Nicholas

**Comment by gordonbazeley** [comment]
No worries.  This result is with a second display connected

Thanks,
  Gordon

**Comment by mgguinne** [comment]
This happens for me on a 2020 Macbook Air M1 (Big Sir) with no touch bar and no external screen connected.

`brightness 0.3
brightness: failed to set brightness of display 0x1 (error -536870201)`

Model Identifier: MacBookAir10,1
Memory: 16 GB
System Firmware Version: 6723.61.3
System Version: macOS 11.1 (20C69)
Kernel Version: Darwin 20.2.0
System Integrity Protection: Enabled

Can't seem to find anything else to work other than using
`osascript -e 'tell application "System Events"' -e 'key code 144' -e ' end tell`

**Comment by nriley** [comment]
Sorry I don’t have a Big Sur or M1 Mac for the moment (or any free time to work on this). If anyone else can figure this out, happy to accept patches.

**Comment by elliottkember** [comment]
Homebrew version gives me the same as above. When compiled directly from source I get:

```
MacBook-Pro ~/Sites/brightness(master) ./brightness -l -v
display 0: main, active, awake, online, built-in, ID 0x1
	resolution 1680 x 1050 pt (3360 x 2100 px) @ 60.0 Hz, origin (0, 0)
	physical size 286 x 179 mm
	IOKit flags 0x3; IOKit display mode ID 0xb
	usable for desktop GUI, uses OpenGL acceleration
display 0: brightness 1.000000
```

But - the `1.000000` is not correct, and never changes regardless of my actual brightness setting.

**Comment by maxgfr** [comment]
+1

**Comment by jtbandes** [comment]
I have a proposed fix: #36

**Comment by dPowNextdoor** [comment]
This is failing for my external monitor (15.6-inch) plugged in via USB-C.

Hardware:

* System: Big Sur (11.4), 2019, 16-inch (built-in) display
* Processor: 2.4 GHz 8-Core Intel Core i9
* Memory: 32 GB 2667 MHz DDR4
* Graphics: AMD Radeon Pro 5300M 4 GB; Intel UHD Graphics 630 1536 MB

Do you have any ideas why, or suggestions for fixes?
